### ISSUE TITLE
Crash on frozen string during force_encoding

### ISSUE THREAD

**Issue by aeris** [trusche/httplog#108]
Hello here!

Since Ruby 2.7 and frozen_string_literal, there is now more and more frozen strings used on Ruby code and gems.
Currently httplog is not compatible with such string, some calls crash if frozen strings met.
At least I spot a crash [here](https://github.com/trusche/httplog/blob/master/lib/httplog/http_log.rb#L335-L339) because `String#force_encoding` modify the string inplace and so raise a `FrozenError: can't modify frozen String: ""` exception.
 
I hit this trouble through https://github.com/mrkamel/swift_client then https://github.com/jnunemaker/httparty then httplog:

```
FrozenError: can't modify frozen String: ""
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/http_log.rb:332:in `force_encoding'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/http_log.rb:332:in `utf_encoded'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/http_log.rb:137:in `parse_body'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/http_log.rb:102:in `log_body'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/http_log.rb:46:in `call'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/adapters/net_http.rb:16:in `request'
  from net/http.rb:1485:in `block in request'
  from net/http.rb:933:in `start'
  from net/http.rb:1483:in `request'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/adapters/net_http.rb:12:in `block in request'
  from benchmark.rb:308:in `realtime'
  from vendor/bundle/ruby/2.7.0/gems/httplog-1.4.3/lib/httplog/adapters/net_http.rb:11:in `request'
  from vendor/bundle/ruby/2.7.0/gems/httparty-0.16.2/lib/httparty/request.rb:143:in `perform'
  from vendor/bundle/ruby/2.7.0/gems/httparty-0.16.2/lib/httparty.rb:563:in `perform_request'
  from vendor/bundle/ruby/2.7.0/gems/httparty-0.16.2/lib/httparty.rb:536:in `head'
  from vendor/bundle/ruby/2.7.0/gems/httparty-0.16.2/lib/httparty.rb:629:in `head'
  from vendor/bundle/ruby/2.7.0/bundler/gems/swift_client-72798338ec47/lib/swift_client.rb:193:in `request'
  from vendor/bundle/ruby/2.7.0/bundler/gems/swift_client-72798338ec47/lib/swift_client.rb:126:in `head_object'
```

I currently bypass the crash with this monkey patch:

```
module Fixtures
  module HttpLog
    def utf_encoded(data, *args)
      data = data.dup if data.frozen?
      super(data, *args)
    end
  end
end

HttpLog.singleton_class.prepend Fixtures::HttpLog
```

Perhaps such protection may be added in other places in the code to be more robust to frozen strings.

**Comment by Meat-Chopper** [comment]
It's fixed in master branch, you can use:
gem 'httplog', github: 'trusche/httplog'

**Comment by trusche** [comment]
I've released the current master as v1.5.0 just now. Please re-open if that doesn't fix it.
